client/blocks/eligibility-warnings/index.tsx

```diff
@@ -38,66 +34,34 @@ interface ExternalProps {
 type Props = ExternalProps & ReturnType< typeof mergeProps > & LocalizeProps;
 
 export const EligibilityWarnings = ( {
-	backUrl,
 	context,
+	feature,
 	eligibilityData,
-	hasBusinessPlan,
 	isEligible,
-	isJetpack,
-	isVip,
 	isPlaceholder,
 	onProceed,
-	onCancel,
+	recordCtaClick,
 	siteId,
-	sitePlan,
-	siteSlug,
 	translate,
 }: Props ) => {
-	const warnings = get( eligibilityData, 'eligibilityWarnings', [] );
-
-	const listHolds = filter(
-		get( eligibilityData, 'eligibilityHolds', [] ),
-		hold => ! includes( [ 'NO_BUSINESS_PLAN', 'NOT_USING_CUSTOM_DOMAIN' ], hold )
-	);
+	const warnings = eligibilityData.eligibilityWarnings || [];
+	const listHolds = eligibilityData.eligibilityHolds || [];
 
+	const showWarnings = warnings.length > 0 && ! hasBlockingHold( listHolds );
 	const classes = classNames( 'eligibility-warnings', {
 		'eligibility-warnings__placeholder': isPlaceholder,
+		'eligibility-warnings--with-indent': showWarnings,
 	} );
 
-	let businessUpsellBanner = null;
-	if ( ! hasBusinessPlan && ! isJetpack && ! isVip ) {
-		const description = translate(
-			'Also get unlimited themes, advanced customization, no ads, live chat support, and more.'
-		);
-		const title = translate( 'Business plan required' ) as string;
-		const plan =
-			sitePlan &&
-			findFirstSimilarPlanKey( sitePlan.product_slug, {
-				type: TYPE_BUSINESS,
-			} );
-		let feature = null;
-		let event = null;
-
-		if ( 'plugins' === context ) {
-			feature = FEATURE_UPLOAD_PLUGINS;
-			event = 'calypso-plugin-eligibility-upgrade-nudge';
-		} else {
-			feature = FEATURE_UPLOAD_THEMES;
-			event = 'calypso-theme-eligibility-upgrade-nudge';
-		}
-
-		const bannerURL = `/checkout/${ siteSlug }/business`;
-		businessUpsellBanner = (
-			<Banner
-				description={ description }
-				feature={ feature }
-				event={ event }
-				plan={ plan }
-				title={ title }
-				href={ bannerURL }
-			/>
-		);
-	}
+	const logEventAndProceed = () => {
+		// We removed <Banner /> Component which logged this event on upsell click.
```

Comment:
I'm not sure this comment is useful in the codebase going forward.  I'd document it in the PR / commit history and take out the comment